### PR TITLE
fix(HMS-1884): return 400 on invalid compose ID

### PR DESCRIPTION
**Pull request by lzap**
We need to return 400 not 500 on invalid compose ID.

Also refactors error handling a bit, it is now more generic and straightforward.

### CODE REVIEW

File: internal/services/aws_reservation_service.go [hunk 1]
```diff
@@ -128,6 +128,7 @@ func CreateAWSReservation(w http.ResponseWriter, r *http.Request) {
 		// Direct AMI or no image were provided (launch template), no need to call image builder
 		ami = reservation.ImageID
 	} else {
+		// Not prefixed with "ami-" therefore this must be a valid UUID
```
Comment (ezr-ondrej) [inline internal/services/aws_reservation_service.go:131]: Could we just include validation here? I mean we do that for all the other providers, so why not here and the feedback for user would be immediate and more accurate, Invalid request - "the imageID must be UUID"

**Review by ezr-ondrej — COMMENTED** [review]
I completely agree with your approach as we can't cover everything image builder validates and we want to catch bad request from the client.

Although in the case we have, I believe we should just validate UUID on our side and give User feedback faster without actually reaching out to IB :thinking:

File: internal/payloads/error_payload.go [hunk 2]
```diff
@@ -94,61 +95,53 @@ func PubkeyDuplicateError(ctx context.Context, message string, err error) *Respo
...
-	if status, message := SourcesErrorHelper(err); status != 0 {
-		return NewResponseError(ctx, status, message, err)
-	}
-	if status, message := ClientErrorHelper(err); status != 0 {
-		return NewResponseError(ctx, status, message, err)
+	if payload := findUserPayload(err); payload != nil {
+		return NewResponseError(ctx, payload.code, payload.message, err)
 	}
-	return NewResponseError(ctx, 500, "HTTP service returned unknown client error", err)
+	log.Ctx(ctx).Warn().Msgf("Unknown client error: %s", err)
```
[the remark this answers is not in the document]
Comment (lzap) [reply internal/payloads/error_payload.go:143]: Good point, I think we should log codes higher than 499 as errors, warnings otherwise.

**Comment by lzap** [comment]
> I completely agree with your approach as we can't cover everything image builder validates and we want to catch bad request from the client.

That is exactly what this patch brings to the table, look for "parse" in the diff. I will just rebase the error vs warning level and we should be goo.

**Comment by ezr-ondrej** [comment]
> That is exactly what this patch brings to the table, look for "parse" in the diff. I will just rebase the error vs warning level and we should be goo.

Ah! my bad, I've completely missed that! :)

**Review by ezr-ondrej — APPROVED** [review]
Thanks @lzap! :+1:

**Pull request merged**